app/controllers/api/enterprises_controller.rb

```diff
@@ -72,5 +76,15 @@ def override_sells
     def override_visible
       params[:enterprise][:visible] = false
     end
+
+    def save_enterprise_users(user_ids)
+      return unless user_ids
+
+      user_ids.each do |user_id|
+        next if @enterprise.user_ids.include? user_id.to_i
+
+        @enterprise.users << Spree::User.find(user_id)
+      end
```

Comment:
I believe we can let ActiveRecord do all this for us (and possibly a bit more efficiently). Looking at the associations Rails guide [collection_singular_ids=(ids)](https://guides.rubyonrails.org/association_basics.html#methods-added-by-has-many-collection-singular-ids-ids) seems to be what we want.

Reply:
I started by trying
@enterprise.user_ids << user_ids

that's what collecction_singular_ids means, right?
but it doesnt work!

Reply:
I don't know if you already did it but this method is a setter so should be used like

```ruby
@enterprise.user_ids = user_ids
```

Reply:
ah, no, maybe that's it :face_with_head_bandage: :see_no_evil:

Reply:
nice, it works. see new commit.
thanks for the lesson!!!